src/components/sqliteStream.js

```diff
@@ -0,0 +1,53 @@
+const Readable = require('stream').Readable;
+const Sqlite3 = require('better-sqlite3');
+
+class SQLiteStream extends Readable {
+  constructor(dbPath, sql) {
+    super({ objectMode: true, autoDestroy: true });
+    this._db = new Sqlite3(dbPath, { readonly: true });
+    this._iterator = this._db.prepare(sql).iterate();
+    this.on('end', () => {
+      this._db.close();
+    });
+  }
+
+  _read() {
+    const elt = this._iterator.next();
+    this.push(!elt.done && elt.value ? elt.value : null);
+  }
+}
+
+/**
+ * Filter deprecated, superseded and null island objects
+ */
+function findGeoJSON() {
+  return `
+SELECT geojson.id, geojson.body
+  FROM geojson JOIN spr
+  ON geojson.id = spr.id
+  WHERE
+    geojson.id != 1
+      AND
+    spr.is_deprecated = 0
+      AND
+    spr.is_superseded = 0
+      AND
+    spr.name <> ''
+      AND
+    (spr.latitude != 0 OR spr.longitude != 0)`;
+}
+
+/**
+ * Add filter and placetypes
+ */
+function findGeoJSONByPlacetype(placetypes) {
+  placetypes = typeof placetypes === 'string' ? [placetypes] : placetypes;
+  return findGeoJSON() + `
+      AND
+    spr.placetype IN ('${placetypes.join('\',\'')}')
+`;
+}
```

Comment:
This function is a `module.export`, so it might pay to be stricter about argument checking and casting here.

With the current code, it's possible to provide a `number` or `undefined` or `null` which will then have `.join()` called on it, potentially resulting in a fatal program error and `exit(1)`.

I personally find the ternary syntax hard to read, wrapping the expression in brackets helps avoid error and make it easier for others to read it.

I would write that line as:
```if( !Array.isArray( placetypes ) ){ placetypes = [ placetypes ]; }```

The advantage of doing it like that is that you'll always end up with an `Array` regardless of the input type, meaning `.join()` won't fail and most types will be passed on verbatim.

Reply:
Updated with more test for this part :+1: